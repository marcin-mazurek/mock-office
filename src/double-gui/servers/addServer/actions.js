export const INIT = 'addServer/INIT';

export default (name, port, serverType, isSecure, keyPath, certPath) => ({
  type: INIT,
  name,
  port,
  serverType,
  isSecure,
  keyPath,
  certPath
});
