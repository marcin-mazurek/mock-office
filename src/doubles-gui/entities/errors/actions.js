export const ADD = 'errors/ADD';
export const REMOVE = 'errors/REMOVE';

export const add = reason => ({
  type: ADD,
  reason
});

export const remove = id => ({
  type: REMOVE,
  id
});
