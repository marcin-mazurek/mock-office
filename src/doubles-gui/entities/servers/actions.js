export const ADD = 'servers/ADD';
export const SELECT = 'servers/SELECT';
export const START = 'servers/START';
export const STOP = 'servers/STOP';
export const REMOVE = 'servers/REMOVE';
export const RENAME = 'servers/RENAME';

export const add = (name, port, serverType, id, secure) => ({
  type: ADD,
  name,
  port,
  id,
  serverType,
  secure
});

export const select = id => ({
  type: SELECT,
  id
});

export const start = id => ({
  type: START,
  id
});

export const stop = id => ({
  type: STOP,
  id
});

export const remove = id => ({
  type: REMOVE,
  id
});

export const rename = (id, name) => ({
  type: RENAME,
  id,
  name
});
