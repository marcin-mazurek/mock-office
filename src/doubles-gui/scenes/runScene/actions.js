export const RUN = 'scenes/RUN';
export const STOP = 'scenes/STOP';

export const run = (serverId, taskId) => ({
  type: RUN,
  serverId,
  taskId
});

export const stop = (serverId, taskId) => ({
  type: STOP,
  serverId,
  taskId
});
