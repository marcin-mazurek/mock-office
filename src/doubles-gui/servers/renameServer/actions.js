export const INIT = 'renameServer/INIT';

export const init = (id, name) => ({
  type: INIT,
  id,
  name
});
