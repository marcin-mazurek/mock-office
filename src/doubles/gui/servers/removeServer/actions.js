export const INIT = 'removeServer/INIT';

export const init = id => ({
  type: INIT,
  id
});
