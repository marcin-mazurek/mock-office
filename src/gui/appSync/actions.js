export const RESTORE_STATE = 'appSync/RESTORE_STATE';
export const restoreStateAction = servers => ({
  type: RESTORE_STATE,
  servers
});
