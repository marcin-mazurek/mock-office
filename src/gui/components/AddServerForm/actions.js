export const SUBMIT_BUTTON_CLICKED = 'AddServerForm/SUBMIT_BUTTON_CLICKED';
export const submitButtonClickAction = values => ({
  type: SUBMIT_BUTTON_CLICKED,
  values
});
