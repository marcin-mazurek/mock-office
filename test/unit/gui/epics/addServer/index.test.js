/* eslint-disable global-require */
import { Map } from 'immutable';
import configureMockStore from 'redux-mock-store';
import { createEpicMiddleware, combineEpics } from 'redux-observable';
import { submitButtonClickAction } from '../../../../../src/gui/components/AddServerForm/actions';
import { SUCCEEDED, FAILED } from '../../../../../src/gui/epics/addServer/actions';

describe('addServerEpic', () => {
  beforeEach(() => {
    jest.resetModules();
    jest.resetAllMocks();
  });

  describe('actions snapshot', () => {
    test('when api resolves promise with server details', (done) => {
      jest.mock('../../../../../src/gui/resources/api', () => ({
        addServer() {
          return Promise.resolve({
            data: {
              id: 'server-id',
              port: 3000,
              name: 'name',
              type: 'http',
              secure: false
            }
          });
        }
      }));
      // eslint-disable-next-line global-require
      const addServerEpic = require('../../../../../src/gui/epics/addServer/index').default;
      let store;
      const helperEpic = action$ =>
        action$.ofType(SUCCEEDED)
          .map(() => {
            expect(store.getActions()).toMatchSnapshot();
            done();
            return undefined;
          });
      const epicMiddleware = createEpicMiddleware(
        combineEpics(addServerEpic, helperEpic)
      );
      const mockStore = configureMockStore([epicMiddleware]);
      store = mockStore();
      const action = submitButtonClickAction(
        new Map({
          port: 3000,
          name: 'name',
          type: 'http',
          secure: false
        })
      );
      store.dispatch(action);
    });

    test('when api rejects promise with error', (done) => {
      jest.mock('../../../../../src/gui/resources/api', () => ({
        addServer() {
          return Promise.resolve({
            error: 'error message'
          });
        }
      }));
      let store;
      // eslint-disable-next-line global-require
      const addServerEpic = require('../../../../../src/gui/epics/addServer/index').default;
      const helperEpic = action$ =>
        action$.ofType(FAILED)
          .map(() => {
            expect(store.getActions()).toMatchSnapshot();
            done();
            return undefined;
          });
      const epicMiddleware = createEpicMiddleware(
        combineEpics(addServerEpic, helperEpic)
      );
      const mockStore = configureMockStore([epicMiddleware]);
      store = mockStore();
      const action = submitButtonClickAction(
        new Map({
          port: 3000,
          name: 'name',
          type: 'http',
          secure: false
        })
      );
      store.dispatch(action);
    });
  });
});
