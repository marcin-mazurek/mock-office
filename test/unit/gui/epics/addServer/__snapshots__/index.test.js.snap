// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addServerEpic actions snapshot when api rejects promise with error 1`] = `
Array [
  Object {
    "type": "AddServerForm/SUBMIT_BUTTON_CLICKED",
    "values": Object {
      "name": "name",
      "port": 3000,
      "secure": false,
      "type": "http",
    },
  },
  Object {
    "reason": "error message",
    "type": "addServer/FAILED",
  },
]
`;

exports[`addServerEpic actions snapshot when api resolves promise with server details 1`] = `
Array [
  Object {
    "type": "AddServerForm/SUBMIT_BUTTON_CLICKED",
    "values": Object {
      "name": "name",
      "port": 3000,
      "secure": false,
      "type": "http",
    },
  },
  Object {
    "params": Object {
      "data": Object {
        "id": "server-id",
        "name": "name",
        "port": 3000,
        "secure": false,
        "type": "http",
      },
    },
    "type": "addServer/SUCCEED",
  },
]
`;
